core/tests/test_utils.py

```diff
@@ -526,6 +526,25 @@ def put_json(self, url, payload, csrf_token=None, expect_errors=False,
         return self._parse_json_response(
             json_response, expect_errors=expect_errors)
 
+    def delete_json(self, url, params, csrf_token=None, expect_errors=False,
+                    expected_status_int=200):
+        """Get an object to the server by JSON; return the received object.
+        Note: The delete method ignores the body of the request. In order
+        overcome this we are attaching the params to the url.
+        """
+        updated_url = url
+        if len(params) > 0:
+            for param in params.values():
+                updated_url += ('/' + param)
+        json_response = self.testapp.delete(
+            updated_url, params, csrf_token, expect_errors=expect_errors,
+            status=expected_status_int)
+        if json_response.content_type == 'application/json':
```

Comment:
When is the response not a JSON? Could you please give an example of a handler that returns a non-JSON response?